### PR TITLE
Adds NOT and XOR triggers for conditions

### LINKED ISSUE #2811: Conditionals: Add support for NOT and XOR in Match
Screen Shot 2020-03-27 at 10 49 39 AM

Add "when no criteria are met" and "when only one criteria is met" respectively.

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #2811 

**Author Checklist**

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? Yes
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/condition/utils/evaluator.js [hunk 1]
```diff
@@ -14,3 +14,19 @@ export const computeCondition = (resultMap, allMustBeTrue) => {
     }
     return result;
 };
+
+//Returns true only if limit number of results are satisfied
+export const computeConditionByLimit = (resultMap, limit) => {
+    let trueCount = 0;
+    for (let key in resultMap) {
+        if (resultMap.hasOwnProperty(key)) {
+            if (resultMap[key]) {
```
Comment (akhenry) [inline src/plugins/condition/utils/evaluator.js]: I think that generally speaking just checking for "truthiness" is a bit vague and can lead to subtle bugs. I'd prefer to be explicit here and check `if (resultMap[key]) === true` or `if (resultMap[key]) !== undefined` - whatever the intended behavior is here.

File: src/plugins/condition/utils/constants.js [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+    ALL: 'all',
+    NOT: 'not',
+    XOR: 'xor'
+};
+
+export const TRIGGER_LABEL = {
+    'any': 'when any criteria are met',
+    'all': 'when all criteria are met',
+    'not': 'when no criteria are met',
+    'xor': 'when only one criteria is met'
```
Comment (akhenry) [inline src/plugins/condition/utils/constants.js]: Being pedantic, but it should be `criterion` singular (as we correctly have it elsewhere in our code).

**Comment by akhenry** [comment]
## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? Y
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

File: src/plugins/condition/utils/evaluator.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/evaluator.js:23]: Done.

File: src/plugins/condition/utils/constants.js (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/constants.js:12]: Done.

**Pull request merged**